### ISSUE TITLE
OSX build fails (TLS version error)

### ISSUE THREAD

**Issue by TheBB** [joerick/cibuildwheel#63]
Hi,

I tried following the instructions [here](https://github.com/joerick/cibuildwheel#minimal-setup), which work fine for Linux, but seem to fail on OSX. Example: https://travis-ci.org/sintefmath/Splipy/jobs/370590848

This seems related to https://github.com/joerick/cibuildwheel/pull/57 which claims to be fixed, so I am sure I'm just doing something wrong here.

.travis.yml file is

```yaml
language: python

matrix:
  include:
    - sudo: required
      services:
        - docker
      env:
        - PIP=pip
        - CIBW_SKIP="cp27-* cp33-*"
    - os: osx
      language: generic
      env:
        - PIP=pip2
        - CIBW_SKIP="cp27-* cp33-*"

script:
  - $PIP install cibuildwheel==0.7.1
  - cibuildwheel --output-dir wheelhouse
```

**Comment by YannickJadoul** [comment]
I do not really see what's going wrong here. Could you maybe try running it again, to see if it wasn't a weird non-reproducible error? - Probably not, but fingers crossed.

Otherwise, I'll have a look at that minimal example, because I have the feeling that it could/should be made simpler anyway...

**Comment by mayeut** [comment]
This seems to be a setuptools issue rather than pip issue which was corrected with #57 
c.f. pypa/setuptools#1320

You can try to install your dependencies using pip before building the wheel (using CIBW_BEFORE_BUILD environment variable)

**Comment by YannickJadoul** [comment]
@TheBB Does this then mean the issue is fixed, given @mayeut's comment?

**Comment by TheBB** [comment]
Sorry, I wasn't able to test it yet. I'll close this for now assuming the fix is good. Thanks everyone.

**Comment by YannickJadoul** [comment]
Thanks! Feel free to re-open if necessary.

**Comment by joerick** [comment]
btw @TheBB, `CIBW_BEFORE_BUILD={pip} install .` is quite common - setuptools installing the `setup_requires` dependencies is pretty unreliable these days.

**Comment by mayeut** [comment]
setuptools is not updated by cibuildwheel and the version installed with python 3.4 on macOS is quite old... Shall cibuildwheel install an up-to-date version of setuptools ? (manylinux images are up-to-date, don't know about windows on AppVeyor)

**Comment by YannickJadoul** [comment]
Hmmm, maybe we should do that. After all, we're also updating `pip` and installing `wheel`...

**Comment by JorjMcKie** [comment]
Similar, but maybe unrelated:
```
[0K$ cibuildwheel --output-dir wheelhouse

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current

                                 Dload  Upload   Total   Spent    Left  Speed


  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
 46 21.4M   46 10.0M    0     0  4808k      0  0:00:04  0:00:02  0:00:02 4807k
100 21.4M  100 21.4M    0     0  9908k      0  0:00:02  0:00:02 --:--:-- 9905k

installer: Package name is Python

installer: Installing at base path /

installer: The install was successful.

/Library/Frameworks/Python.framework/Versions/2.7/bin/python2

Python 2.7.13

Requirement already up-to-date: setuptools in /Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages

Requirement already up-to-date: pip in /Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages

pip 9.0.1 from /Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages (python 2.7)

Collecting wheel

  Could not fetch URL https://pypi.python.org/simple/wheel/: There was a problem confirming the ssl certificate: [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:661) - skipping

  Could not find a version that satisfies the requirement wheel (from versions: )

No matching distribution found for wheel

```
I am now going to try excluding Python versions below 3 altogether ...

**Comment by joerick** [comment]
Fixed in #71, released in 0.9.1!
